launch/launch/substitutions/string_join_substitution.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2025 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the StringJoinSubstitution substitution."""
+
+from typing import Iterable, List, Text
+
+from ..launch_context import LaunchContext
+from ..some_substitutions_type import SomeSubstitutionsType
+from ..substitution import Substitution
+from ..utilities import perform_substitutions
+
+
+class StringJoinSubstitution(Substitution):
+    """
+    Substitution that joins strings and/or other substitutions.
+
+    This takes in a list of string components as substitutions.
+    The substitutions for each string component are performed and concatenated,
+    and then all string components are joined with a specified delimiter as seperation.
+
+    For example:
+
+    .. code-block:: python
+
+        StringJoinSubstitution(
+            ['https://', LaunchConfiguration('subdomain')], 'ros', 'org'],
+            delimiter='.'
+        )
```

Comment:
I would augment this example (without changing the resulting string) to show that each component can be a list that gets concatenated, similar to some of your tests and similar to the example I added in #838 + https://github.com/ros2/ros2_documentation/pull/5157 (see #835).

Maybe something simple, like splitting `'https://'` in 2:

```suggestion
        StringJoinSubstitution(
            [['https', '://'], LaunchConfiguration('subdomain')], 'ros', 'org'],
            delimiter='.'
        )
```